doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,157 +1257,155 @@ Managing the local context
 
-   .. tacv:: move @ident at top
-      :name: move … at top
+      .. todo This is dreadful behavior
 
-      This moves :token:`ident` at the top of the local context (at the beginning
-      of the context).
+   .. exn:: Cannot move @ident__from after @ident: it occurs in the type of @ident.
+      :undocumented:
 
-   .. tacv:: move @ident at bottom
-      :name: move … at bottom
+   .. exn:: Cannot move @ident__from after @ident: it depends on @ident.
+      :undocumented:
 
-      This moves :token:`ident` at the bottom of the local context (at the end of
-      the context).
+   .. example:: move
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+      .. coqtop:: reset none
 
-   .. exn:: Cannot move @ident__1 after @ident__2: it occurs in the type of @ident__2.
-      :undocumented:
+         Goal forall x :nat, x = 0 -> forall y z:nat, y=y-> 0=x.
 
-   .. exn:: Cannot move @ident__1 after @ident__2: it depends on @ident__2.
-      :undocumented:
+      .. coqtop:: out
 
-   .. example::
+           intros x Hx y z Hy.
 
-      .. coqtop:: reset all
+      .. coqtop:: in
 
-         Goal forall x :nat, x = 0 -> forall z y:nat, y=y-> 0=x.
-           intros x H z y H0.
-           move x after H0.
+           (*                    x Hx y z Hy *)
+           move y after z.  (*    x Hx z y Hy   (z was left of y, intuitive case) *)
            Undo.
-           move x before H0.
+           move z after y.  (*    x Hx z y Hy   (z was right of y, see Note above) *)
            Undo.
-           move H0 after H.
+           move x after Hy.  (*   y z Hy x Hx   (Hx depends on x, so moved) *)
            Undo.
-           move H0 before H.
+           move x before Hy.  (*  y z x Hx Hy *)
+           Undo.
+           move Hy after Hx.  (*  x y Hy Hx z *)
+           Undo.
+           move Hy before Hx.  (* x Hx y Hy z *)
 
-.. tacn:: rename @ident__1 into @ident__2
-   :name: rename
+.. tacn:: rename {+, @ident__1 into @ident__2 }
 
-   This renames hypothesis :n:`@ident__1` into :n:`@ident__2` in the current
-   context. The name of the hypothesis in the proof-term, however, is left
-   unchanged.
+   Renames hypothesis :n:`@ident__1` into :n:`@ident__2` for each pair of :n:`@ident`\s.
+   Renaming is done simultaneously, which permits swapping the names of 2 hypotheses.
+   (Note that the renaming is applied in the context and the existential
+   variables, but the proof term doesn't change.)
 
-   .. tacv:: rename {+, @ident__i into @ident__j}
+   .. exn:: @ident is already used.
+      :undocumented:
 
-      This renames the variables :n:`@ident__i` into :n:`@ident__j` in parallel.
-      In particular, the target identifiers may contain identifiers that exist in
-      the source context, as long as the latter are also renamed by the same
-      tactic.
+.. tacn:: set @one_term {? @as_name } {? @occurrences }
+          set @bindings_with_parameters {? @occurrences }
+   :name: set; _
```

Comment:
Actually, I suggest reversing the order of the two forms. The second one is the most generic and most common one and the second one could be described as just a shorthand for `set (@ident := @one_term) {? @occurrences }` (where `@ident` is auto-generated in case it was not provided).

Reply:
I think I got all you suggestions, please take a look at the result.

Reply:
Looks perfect!